### DISCUSSION TITLE
Is there a difference between page.set_rotation(0) and page.remove_rotation() ?

### DISCUSSION THREAD

**Discussion by cmyers009** [pymupdf/PyMuPDF#3638] [Q&A] [answered]
I saw that the function page.remove_rotation() was added in 1.24.3.

I have been using page.set_rotation(0) for a similar purpose prior to this function addition.

In the change-log I see that the description says:
new, set page rotation to zero while keeping appearance.

Does this mean that page.set_rotation(0) has the potential to affect the appearance of the page,
or is page.remove_rotation() just short-hand for page.set_rotation(0) ?

**Comment by JorjMcKie** [comment, marked as answer]
Absolutely!
`Page.set-rotation()` **_certainly_** changes the appearance! E.g. with the value 90, everything will appear rotated clockwise.
The attractive benefit of `Page.remove_rotation()` is that, given a page rotated by 90° clockwise and text being written on it in a way that is readable (= it has been written rotated by -90° relative to that page) in the normal way, top-left to bottom-right.
If you set the rotation of that page to 0, text will appear written from bottom-left to top-right. I.e. from:

![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/46117661-64a8-443f-a730-ca61aa5afd01)

to

![image](https://github.com/pymupdf/PyMuPDF/assets/8290722/74b4f321-2829-4bac-a5f9-0cdfbf1c6da2)

`Page.remove_rotation()` prevent this from happening. The page will have rotation 0, but an exchanged width / height and no visual differences.
